Core.UnitTests/MutableReflection/Implementation/InterfaceMappingComputerTest.cs

```diff
@@ -185,44 +211,51 @@ public void ComputeMapping_AddedInterface_Candidates_AllowPartial ()
-        "'NonPublicMethod', 'NonVirtualMethod'.")]
     public void ComputeMapping_AddedInterface_Candidates_Throws ()
     {
       _mutableType.AddInterface (typeof (IImplementationCandidates));
-      _computer.ComputeMapping (_mutableType, _interfaceMapProviderMock.Object.Get, typeof (IImplementationCandidates), false);
+      Assert.That (
+          () => _computer.ComputeMapping (_mutableType, _interfaceMapProviderMock.Object.Get, typeof (IImplementationCandidates), false),
+          Throws.InvalidOperationException
+              .With.Message.EqualTo (
+                  "The added interface 'IImplementationCandidates' is not fully implemented. The following methods have no implementation: " +
+                  "'NonPublicMethod', 'NonVirtualMethod'."));
     }
 
     [Test]
-    [ExpectedException (typeof (ArgumentException), ExpectedMessage = "Type passed must be an interface.\r\nParameter name: interfaceType")]
     public void ComputeMapping_NoInterfaceType ()
     {
-      _computer.ComputeMapping (_mutableType, _interfaceMapProviderMock.Object.Get, typeof (object), false);
+      Assert.That (
+          () => _computer.ComputeMapping (_mutableType, _interfaceMapProviderMock.Object.Get, typeof (object), false),
+          Throws.ArgumentException
+              .With.Message.EqualTo (
+                  "Type passed must be an interface.\r\nParameter name: interfaceType"));
     }
 
     [Test]
-    [ExpectedException (typeof (ArgumentException), ExpectedMessage = "Interface not found.\r\nParameter name: interfaceType")]
     public void ComputeMapping_NotImplemented ()
     {
-      _computer.ComputeMapping (_mutableType, _interfaceMapProviderMock.Object.Get, typeof (IDisposable), false);
+      Assert.That (
+          () => _computer.ComputeMapping (_mutableType, _interfaceMapProviderMock.Object.Get, typeof (IDisposable), false),
+          Throws.ArgumentException
+              .With.Message.EqualTo (
+                  "Interface not found.\r\nParameter name: interfaceType"));
     }
 
-    // Tuple means: 1) interface method, 2) implementation method
-    private void CallComputeMappingAndCheckResult (MutableType mutableType, Type interfaceType, params Tuple<MethodInfo, MethodInfo>[] expectedMapping)
+    private void CallComputeMappingAndCheckResult (MutableType mutableType, Type interfaceType, params MethodMapping[] expectedMapping)
     {
       var mapping = _computer.ComputeMapping (mutableType, _interfaceMapProviderMock.Object.Get, interfaceType, true);
 
       _interfaceMapProviderMock.Verify();
       Assert.That (mapping.InterfaceType, Is.SameAs (interfaceType));
       Assert.That (mapping.TargetType, Is.SameAs (mutableType));
       // Order matters for "expectedMapping".
-      var enumerable = mapping.InterfaceMethods.Zip (mapping.TargetMethods, Tuple.Create).ToArray();
+      var enumerable = mapping.InterfaceMethods.Zip (mapping.TargetMethods, (a, b) => new MethodMapping(a, b)).ToArray();
```

Comment:
Just to make sure, ValueTuple also doesn't work here? Because otherwise, we could just go with ValueTuple to simplify the code changes.

Reply:
`ValueTuple` should work, I haven't tried because it was not referenced yet. But I think that that the current solution is better than a `ValueTuple` where you have to guess what side means what. With records the `MethodMapping` class will be a one-liner.

Reply:
No, you can use named parameters with ValueTuple. No guesswork involved but less code.

Reply:
I tried using `ValueType` now and it does not work. It throws the same exception as the original `Tuple<>` code. The whole thing is just very strange for no reason. `EquivalentTo` seems to hate tuples.

Reply:
Okay, so be it.

Say, should we report this to Nunit with the option of contribbing?

Reply:
It is not an NUnit bug. NUnit 3 checks by equivalence by sorting the array. This requires `IComparable`. `ValueTuple` implements `IComparable` but because it won't know what types are supplied it checks for comparability when `Compare` is called. `MethodInfo` is not comparable, which is why the test fails.

The custom class is not comparable but equatable, which falls back to another way of checking equivalence in NUnit. All in all very unfortunate for our scenario...